sim/mage/flame_orb.go

```diff
@@ -0,0 +1,160 @@
+package mage
+
+import (
+	"time"
+
+	"github.com/wowsims/cata/sim/core"
+	"github.com/wowsims/cata/sim/core/stats"
+)
+
+func (mage *Mage) registerFlameOrbSpell() {
+
+	// Frostfire Orb talent converts Flame Orb spell into Frostfire Orb spell.
+	// Don't allow a user access to both at the same time.
+	if mage.Talents.FrostfireOrb != 0 {
+		return
+	}
+
+	mage.FlameOrb = mage.RegisterSpell(core.SpellConfig{
+		ActionID:       core.ActionID{SpellID: 82731},
+		SpellSchool:    core.SpellSchoolFire,
+		ProcMask:       core.ProcMaskEmpty, //tbd
+		Flags:          SpellFlagMage | core.SpellFlagAPL,
+		ClassSpellMask: MageSpellFlameOrb,
+
+		ManaCost: core.ManaCostOptions{
+			BaseCost: 0.06,
+		},
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				GCD: core.GCDDefault,
+			},
+			CD: core.Cooldown{
+				Timer:    mage.NewTimer(),
+				Duration: time.Minute,
+			},
+		},
+
+		ApplyEffects: func(sim *core.Simulation, _ *core.Unit, _ *core.Spell) {
+			mage.flameOrb.EnableWithTimeout(sim, mage.flameOrb, time.Second*15)
+		},
+	})
+
+	mage.AddMajorCooldown(core.MajorCooldown{
+		Spell: mage.FlameOrb,
+		Type:  core.CooldownTypeDPS,
+	})
+}
+
+func (mage *Mage) registerFlameOrbExplodeSpell() {
+
+	mage.FlameOrbExplode = mage.RegisterSpell(core.SpellConfig{
+		ActionID:       core.ActionID{SpellID: 83619},
+		SpellSchool:    core.SpellSchoolFire,
+		ProcMask:       core.ProcMaskSpellDamage | core.ProcMaskNotInSpellbook,
+		Flags:          SpellFlagMage, // | HotStreakSpell (unlikely based on videos)
+		ClassSpellMask: MageSpellFlameOrb,
+
+		DamageMultiplier: 1,
+		CritMultiplier:   mage.DefaultSpellCritMultiplier(),
+		BonusCoefficient: 0.193,
+		ThreatMultiplier: 1,
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+
+			damage := 1.318 * mage.ScalingBaseDamage
+
+			for _, aoeTarget := range sim.Encounter.TargetUnits {
+				spell.CalcAndDealDamage(sim, aoeTarget, damage, spell.OutcomeMagicHitAndCrit)
+				spell.SpellMetrics[target.UnitIndex].Hits++
```

Comment:
Why are you manually increasing hit metrics here?

Reply:
idk, that was dumb.